File: app/src/test/java/info/nightscout/androidaps/interfaces/PluginDescriptionTest.java
package info.nightscout.androidaps.interfaces;

import org.junit.Assert;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.powermock.modules.junit4.PowerMockRunner;

import info.nightscout.androidaps.plugins.profile.ns.NSProfileFragment;

@RunWith(PowerMockRunner.class)
public class PluginDescriptionTest {

    @Test
    public void mainTypeTest() {
        PluginDescription pluginDescription = new PluginDescription().mainType(PluginType.PUMP);
        Assert.assertEquals(PluginType.PUMP, pluginDescription.getType());
    }

    @Test
    public void fragmentClassTest() {
        PluginDescription pluginDescription = new PluginDescription().fragmentClass(NSProfileFragment.class.getName());
        Assert.assertEquals(NSProfileFragment.class.getName(), pluginDescription.getFragmentClass());
    }

    @Test
    public void alwaysEnabledTest() {
        PluginDescription pluginDescription = new PluginDescription().alwaysEnabled(true);
        Assert.assertEquals(true, pluginDescription.alwaysEnabled);
    }

    @Test
    public void alwayVisibleTest() {
        PluginDescription pluginDescription = new PluginDescription().alwaysVisible(true);
        Assert.assertEquals(true, pluginDescription.alwaysVisible);
    }

    @Test
    public void neverVisibleTest() {
        PluginDescription pluginDescription = new PluginDescription().neverVisible(true);
        Assert.assertEquals(true, pluginDescription.neverVisible);
    }

    @Test
    public void showInListTest() {
        PluginDescription pluginDescription = new PluginDescription().showInList(false);
        Assert.assertEquals(false, pluginDescription.showInList);
    }

    @Test
    public void pluginName() {
        PluginDescription pluginDescription = new PluginDescription().pluginName(10);
        Assert.assertEquals(10, pluginDescription.pluginName);
    }

    @Test
    public void shortNameTest() {
        PluginDescription pluginDescription = new PluginDescription().shortName(10);
        Assert.assertEquals(10, pluginDescription.shortName);
    }

    @Test
    public void preferencesIdTest() {
        PluginDescription pluginDescription = new PluginDescription().preferencesId(10);
        Assert.assertEquals(10, pluginDescription.preferencesId);
    }

    @Test
    public void enableByDefault() {
        PluginDescription pluginDescription = new PluginDescription().enableByDefault(true);
        Assert.assertEquals(true, pluginDescription.enableByDefault);
    }

    @Test
    public void visibleByDefault() {
        PluginDescription pluginDescription = new PluginDescription().visibleByDefault(true);
        Assert.assertEquals(true, pluginDescription.visibleByDefault);
    }

}

File: app/src/test/java/info/nightscout/androidaps/interfaces/PluginDescriptionTest.kt
package info.nightscout.androidaps.interfaces

import info.nightscout.androidaps.plugins.profile.ns.NSProfileFragment
import org.junit.Assert
import org.junit.Test
import org.junit.runner.RunWith
import org.powermock.modules.junit4.PowerMockRunner

@RunWith(PowerMockRunner::class)
class PluginDescriptionTest {

    @Test fun mainTypeTest() {
        val pluginDescription = PluginDescription().mainType(PluginType.PUMP)
        Assert.assertEquals(PluginType.PUMP, pluginDescription.type)
    }

    @Test fun fragmentClassTest() {
        val pluginDescription = PluginDescription().fragmentClass(NSProfileFragment::class.java.name)
        Assert.assertEquals(NSProfileFragment::class.java.name, pluginDescription.getFragmentClass())
    }

    @Test fun alwaysEnabledTest() {
        val pluginDescription = PluginDescription().alwaysEnabled(true)
        Assert.assertEquals(true, pluginDescription.alwaysEnabled)
    }

    @Test fun alwayVisibleTest() {
        val pluginDescription = PluginDescription().alwaysVisible(true)
        Assert.assertEquals(true, pluginDescription.alwaysVisible)
    }

    @Test fun neverVisibleTest() {
        val pluginDescription = PluginDescription().neverVisible(true)
        Assert.assertEquals(true, pluginDescription.neverVisible)
    }

    @Test fun showInListTest() {
        val pluginDescription = PluginDescription().showInList(false)
        Assert.assertEquals(false, pluginDescription.showInList)
    }

    @Test fun pluginIcon() {
        val pluginDescription = PluginDescription().pluginIcon(10)
        Assert.assertEquals(10, pluginDescription.pluginIcon.toLong())
    }

    @Test fun pluginName() {
        val pluginDescription = PluginDescription().pluginName(10)
        Assert.assertEquals(10, pluginDescription.pluginName.toLong())
    }

    @Test fun shortNameTest() {
        val pluginDescription = PluginDescription().shortName(10)
        Assert.assertEquals(10, pluginDescription.shortName.toLong())
    }

    @Test fun preferencesIdTest() {
        val pluginDescription = PluginDescription().preferencesId(10)
        Assert.assertEquals(10, pluginDescription.preferencesId.toLong())
    }

    @Test fun enableByDefault() {
        val pluginDescription = PluginDescription().enableByDefault(true)
        Assert.assertEquals(true, pluginDescription.enableByDefault)
    }

    @Test fun visibleByDefault() {
        val pluginDescription = PluginDescription().visibleByDefault(true)
        Assert.assertEquals(true, pluginDescription.visibleByDefault)
    }
}